### PR TITLE
Fix arch handling in chip_xcode_build_connector

### PR DESCRIPTION
**Pull request by ksperling-apple**
Set target_cpu based on first arch in ARCHS so Xcode builds for Mac work on M1 Macs.

`PLATFORM_PREFERRED_ARCH` seems to be independent of the host architecture, so we end up with the wrong `target_cpu` and `BUILDCONFIG.gn` doesn't find the tool chain.

An alternative fix might be to tweak the check in `BUILDCONFIG.gn` that selects the host tool chain (`target_os == host_os && target_cpu == host_cpu`) to ignore the CPU type when building on Mac for Mac. (It's not entirely clear to me what `target_cpu` is meant to be when building for both `arm64` and `x86_64` at the same time.)

Also tweak project / scheme settings.

### CODE REVIEW

**Comment by ksperling-apple** [comment]
There's also this ```'mac_target_arch="'"$PLATFORM_PREFERRED_ARCH"'"'```, not sure if that should change too.

File: src/darwin/Framework/chip_xcode_build_connector.sh [hunk 1]
```diff
@@ -61,29 +61,14 @@ done
...
-        ;;
-esac
-
 declare target_cflags=""
 
 read -r -a archs <<<"$ARCHS"
 
 for arch in "${archs[@]}"; do
+    if [ -z "$target_cpu"] ]; then
+        target_cpu="$arch"
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/chip_xcode_build_connector.sh:70]: The case statement being removed above had different values for PLATFORM_PREFERRED_ARCH and `target_cpu`.  Are the values in ARCHS not the same as the values that can end up in PLATFORM_PREFERRED_ARCH?

File: src/darwin/Framework/chip_xcode_build_connector.sh [hunk 2]
```diff
@@ -165,5 +150,5 @@ find_in_ancestors() {
 
     # gnerate and build
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/chip_xcode_build_connector.sh:151]:
```suggestion
    # generate and build
```
Yes, I know, pre-existing.

File: src/darwin/Framework/chip_xcode_build_connector.sh (hunk 1 above)
Comment (ksperling-apple) [reply src/darwin/Framework/chip_xcode_build_connector.sh:70]: Yes, I probably need to handle `x86_64` -> `x64`. I don't think `armv7` and `i386` are actually relevant for Darwin builds?

Comment (bzbarsky-apple) [reply src/darwin/Framework/chip_xcode_build_connector.sh:70]: Probably not, yeah.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**